src/bugreporter/odemis_bugreporter.py

```diff
@@ -739,6 +779,81 @@ def open_failed_upload_dlg(self, error: str):
             webbrowser.open('https://support.delmic.com/open.php')
 
 
+    def _add_attachment_dialog(self, evt: wx.Event) -> List[str]:
+        """Open a dialog to select files to attach to the bug report"""
+
+        dlg = wx.FileDialog(self, message="Select files to attach",
+                            wildcard="All files (*.*)|*.*",
+                            style=wx.FD_OPEN | wx.FD_MULTIPLE)
+
+        if dlg.ShowModal() == wx.ID_OK:
+            paths = dlg.GetPaths()
+        else:
+            paths = []
+        dlg.Destroy()
+
+        # Add the selected files to the list of attachments
+        self.bugreporter.user_attachments.extend(paths)
+        logging.info("Added attachments: %s", paths)
+
+        # update label
+        self._update_attachment_label()
+
+    def _remove_attachments(self, evt: wx.Event) -> None:
+        """Remove all user-selected attachments"""
+        self.bugreporter.user_attachments.clear()
+        logging.info("Removed all attachments")
+
+        # update label
+        self._update_attachment_label()
+
+    def _update_attachment_label(self) -> None:
+        """Update the label showing the number of attached files"""
+
+        paths = self.bugreporter.user_attachments
+
+        # no files attached
+        if len(paths) == 0:
+            self.attached_files_lbl.SetForegroundColour(wx.Colour(255, 255, 255))
+            self.attached_files_lbl.SetLabel("No files attached")
+            return
+
+        valid_attachments = self._validate_user_attachment_size()
+        self.report_btn.Enable(valid_attachments)
+
+        if not valid_attachments:
+            max_str = int(MAX_ATTACHMENT_SIZE / (1024*1024))
+            self.attached_files_lbl.SetLabel(f"Total size of attachments exceeds the limit ({max_str}MB). Please remove some attached files.")
+            # set text color to red
+            self.attached_files_lbl.SetForegroundColour(wx.Colour(255, 0, 0))
+            self.Layout()
+            return
+
+        # show the number of attached files
+        if len(paths) == 1:
+            label = "1 file attached"
+        else:
+            label = f"{len(paths)} files attached"
+
+        # resize the label to fit the text
+        self.attached_files_lbl.Wrap(self.attached_files_lbl.GetSize().width)
+        self.Layout()
+
+        # show each attachment basename on a new line
+        self.attached_files_lbl.SetForegroundColour(wx.Colour(255, 255, 255))
+        label = f"{label}\n" + "\n".join(os.path.basename(path) for path in paths)
+        self.attached_files_lbl.SetLabel(label)
+
+    def _validate_user_attachment_size(self) -> bool:
+        """Check if the total size of the attached files is below the maximum limit"""
+
+        total_size = sum(os.path.getsize(path) for path in self.bugreporter.user_attachments)
+        if total_size > MAX_ATTACHMENT_SIZE:
+            logging.warning("Total size of user attachments exceeds the limit")
```

Comment:
Please add the total_size value in the log message (and MAX_ATTACHMENT_SIZE would probably be handy too).